### PR TITLE
Find a roomate component

### LINKED ISSUE #12: Define an API call
As a dev,
I would like to have a function to call the residents api that can be invoked in the Form component.

### PR DESCRIPTION
**Pull request by cameronRomo**
### What’s this PR do?

* Adds an api call to get resident data and passes it down to the Thumbnails 

### Where should the reviewer start?

* `src/App/App.js` line 2
* `src/FindARoommate/FindARoommate.js` line 1
* `src/Form/Form.js` line 1
* `src/NavBar/NavBar.js` line 32
* `src/Thumbnails/Thumbnails.js` line 32

### How should this be manually tested?

* Pull down launch site and notice the residents render in the scroll bar

### Any background context you want to provide?

* Heavy on react with hooks and api calls

### Screenshots:
Screen Shot 2021-01-09 at 11 22 27 PM

### Relevent Tickets

* Closes #12

### CODE REVIEW

File: src/FindARoommate/FindARoommate.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  const selectSemester = (selectedSemester) => {
+    setSemester(selectedSemester)
+  }
+  
+  return(
+    <div>
+      <ControlledOpenSelect selectSemester={selectSemester}/>
+      <SingleLineGridList availableResidents={residents} semesterAvailable={semester}/>
```
Comment (lo-la-do-li) [inline src/FindARoommate/FindARoommate.js:28]: We can change this prop (residents) to residentsAvailable once we're able to query residents on our server by the semester property

Comment (chadrick-d-dev) [inline src/FindARoommate/FindARoommate.js:28]: glad we remembered to pass use the variable name as the prop. lol :D

File: src/Thumbnails/Thumbnails.js [hunk 2]
```diff
@@ -29,57 +29,57 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-const tileData = [
+// const residents = [
```
Comment (chadrick-d-dev) [inline src/Thumbnails/Thumbnails.js:32]: Never forget! 🕯

File: src/Form/Form.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+          open={open}
+          onClose={handleClose}
+          onOpen={handleOpen}
+          value={semester}
+          onChange={handleChange}
+        >
+          <MenuItem value="">
+            <em>None</em>
+          </MenuItem>
+          <MenuItem value='Spring 2021'>Spring 2021</MenuItem>
```
Comment (lo-la-do-li) [inline src/Form/Form.js:50]: We'll need this MenuItem value probably as an argument to pass to our apiCall. Looking good - we got our data to render! Woohoo!

File: src/Thumbnails/Thumbnails.js [hunk 4]
```diff
@@ -29,57 +29,57 @@ const useStyles = makeStyles((theme) => ({
...
  
-  {
```
Comment (lo-la-do-li) [inline src/Thumbnails/Thumbnails.js:34]: We can delete this now

**Review by chadrick-d-dev — APPROVED** [review]
So much accomplished! Great Job Team!

**Pull request merged**